### ISSUE TITLE
Plugin can erase all other plugins in iOS preventing app from loading

### ISSUE THREAD

**Issue by ilyakamens** [BranchMetrics/cordova-ionic-phonegap-branch-deep-linking#516]
The issue stems from the fact that `beforePrepare.js` is executed in both `after_plugin_install` and `before_prepare` and caches project files.

Here's a detailed walkthrough of how this happens:
1. User adds Branch Cordova library to `config.xml`
2. `cordova platform add ios`
3. `beforePrepare.js` (which reads + modifies `<project>.xcodeproj`) runs in `after_plugin_install`
4. Other plugins are installed and they are added to `<project>.xcodeproj`
5. `before_prepare` hooks are run and `beforePrepare.js` runs again

Here's the call chain of `beforePrepare.js` that causes things to break:
1. https://github.com/BranchMetrics/cordova-ionic-phonegap-branch-deep-linking/blob/master/src/scripts/hooks/beforePrepare.js#L15
2. https://github.com/BranchMetrics/cordova-ionic-phonegap-branch-deep-linking/blob/master/src/scripts/npm/processConfigXml.js#L17
3. https://github.com/BranchMetrics/cordova-ionic-phonegap-branch-deep-linking/blob/master/src/scripts/npm/processConfigXml.js#L64
4. https://github.com/BranchMetrics/cordova-ionic-phonegap-branch-deep-linking/blob/master/src/scripts/npm/processConfigXml.js#L142
5. https://github.com/apache/cordova-ios/blob/master/bin/templates/scripts/cordova/lib/projectFile.js#L62

As you can see, the project files are cached; as a result, `<project>.xcodeproj` is cached when read in `after_plugin_install` before any other plugins have been added, and then the file which now has other plugin info is overwritten by the cached version when `beforePrepare.js` is run in `before_prepare`.

There are two fixes:
1. Don't run `beforePrepare.js` in `after_plugin_install`. This is what I've done in my forked version of this repo, since I don't see why `beforePrepare.js` needs to be run twice.
2. Purge the cache each time via `purgeProjectFileCache` https://github.com/apache/cordova-ios/blob/master/bin/templates/scripts/cordova/lib/projectFile.js#L94

Please let me know which fix you'd prefer and I'd be happy to submit a pull request.

**Comment by Sarkar** [comment]
@sequoiaat, what are your thoughts on the two proposed solutions?

@ilyakamens, thank you for the incredible detail. Intuitively, I'd think we'd need to do #1, but I'd wanna validate that this doesn't break anything else (I believe it doesn't). #2 seems like a safer method. Will wait for @sequoiaat to test and confirm, given the two options.

**Comment by sequoiaat** [comment]
@Sarkar  We will test them out and give our thoughts next week

**Comment by sequoiaat** [comment]
@sarkar we tried out 1# and everything seems working fine and did not see any unexpected behaviors.

**Comment by Sarkar** [comment]
Fantastic. Let's do #1. @sequoiaat will you be able to open a PR today? @ilyakamens, alternatively, we can just review your fork and bring it back into master here. Let me know what's easier.

**Comment by ilyakamens** [comment]
Here it is: https://github.com/propelinc/cordova-ionic-phonegap-branch-deep-linking/commit/aa478f93f78163687b14facb7111f032ff23ca0d

**Comment by Sarkar** [comment]
Thanks for the help, @ilyakamens. we'll deploy shortly - if you just pull master you should be good.

**Comment by ljudbane** [comment]
This is still an issues in version 3.2.0. When i remove ios platform and then add it, only Branch's files get added to ios project and not from other plugins.
I noticed that fix from this issue (propelinc@aa478f9) was reverted shortly after and beforePrepare.js hook gets called twice again. If i remove after_plugin_install hook from plugin.xml then it works fine.

I'm using older version of cordova so maybe this is specific to my version:
cordova (Cordova CLI) : 6.5.0
Cordova Platforms     : android 6.3.0, ios 4.5.5
